### ISSUE TITLE
jq is compiled without ONIGURUMA regex libary

### ISSUE THREAD

**Issue by smetj** [mwilliamson/jq.py#18]
```python
>>> jq('.["one"] | test( ".*") ').transform({"one": 1})
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "jq.pyx", line 123, in jq._Program.transform (jq.c:1766)
ValueError: jq was compiled without ONIGURUMA regex libary. match/test/sub and related functions are not available.
>>> 
```

**Comment by smetj** [comment]
After installing the oniguruma-devel libs reinstalling jq returns following error:

 ```python
Python 2.7.11 (default, Dec 13 2015, 12:30:38) 
[GCC 5.1.1 20150618 (Red Hat 5.1.1-4)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> from jq import jq
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: /home/smetj/data/python/virtualenv/p/lib/python2.7/site-packages/jq.so: undefined symbol: OnigSyntaxPerl_NG
```

**Comment by mwilliamson** [comment]
I suspect jq is not including Oniguruma when building the static library that the Python bindings rely on. Could you try cloning this repository and running `make test` on the branches `master` and `static-jq`?

**Comment by mwilliamson** [comment]
This issue in the jq repository might also be helpful in debugging: https://github.com/stedolan/jq/issues/613

**Comment by smetj** [comment]
I got a working albeit hacky version ready which includes ONIGURAMA and statically links it ...

https://github.com/smetj/jq.py/blob/master/setup.py

There is a nasty hack in there, (like line 63) ... this isn' t required when installing your original package. I'm not sure which part of the setup normally is responsible for doing this ...

Perhaps you could include some of these changes in your library so onigurama is included (it's an indispensable part imo)

**Comment by mwilliamson** [comment]
Thanks for taking a look! As you've already alluded to, that branch probably needs a few changes before it can be merged in. Would you rather I give some feedback and let you make the changes, or shall I make them myself?

**Comment by smetj** [comment]
Hi,

Well, If you don't mind, perhaps it'll be the most efficient if you do the changes yourself ...

Thanks a million!

**Comment by mwilliamson** [comment]
Would you mind taking a look at the `oniguruma` branch and see if it works for you? Note that the repository is not meant to be directly installable -- you need to run `cython jy.pyx` before attempting to install it.

**Comment by smetj** [comment]
I get the following error:
```
running install_lib
running build_ext
Executing: ./configure CFLAGS=-fPIC --prefix=onig-install-5.9.6
configure: error: expected an absolute directory name for --prefix: onig-install-5.9.6
Traceback (most recent call last):
...snip...
subprocess.CalledProcessError: Command '['./configure', 'CFLAGS=-fPIC', '--prefix=onig-install-5.9.6']' returned non-zero exit status 1

```

**Comment by mwilliamson** [comment]
I've just pushed an update, could you try again?

**Comment by smetj** [comment]
Works like a charm!

```
>>> from jq import jq
>>> jq('.test | test ( "blah" )').transform({"test": "blah"})
True
>>> 

```

**Comment by smetj** [comment]
Somewhat unrelated, but could you explain which mechanism is responsible for running:

`$cython jq.pyx`

When installing via pip?

**Comment by mwilliamson** [comment]
None, the source distribution includes the `.c` file rather than the `.pyx` file.

**Comment by smetj** [comment]
hah, I feel somewhat disappointed :)

Are you planning a new release for this ?

**Comment by mwilliamson** [comment]
Yes, although it'd be good to get people on other platforms to try it out.

**Comment by mwilliamson** [comment]
I've just released the changes as 0.1.6. Thanks again for taking a look at this!

**Comment by smetj** [comment]
awesome thank you!
